### DISCUSSION TITLE
Is there a way to use Explain with sqlalchemy? How to check performance with sqlalchemy

### DISCUSSION THREAD

**Discussion by martinenkoEduard** [sqlalchemy/sqlalchemy#8245] [Usage Questions] [answered]
Is there a way to use Explain with sqlalchemy? How to check performance with sqlalchemy?

**Comment by CaselIT** [comment, marked as answer]
Hi,

not directly, if you have a text sql you can just add explain to it. Otherwise you can use the `before_cursor_execute` event to insert explain to the beginning of the query string produced by sqlalchemy. See https://docs.sqlalchemy.org/en/14/core/events.html?highlight=before+execute#sqlalchemy.events.ConnectionEvents.before_cursor_execute
